sklearn/metrics/_regression.py

```diff
@@ -137,6 +140,63 @@ def _check_reg_targets(y_true, y_pred, multioutput, dtype="numeric", xp=None):
     return y_type, y_true, y_pred, multioutput
 
 
+def _check_reg_targets_with_floating_dtype(
+    y_true, y_pred, sample_weight, multioutput, xp=None
+):
+    """Ensure that y_true and y_pred correspond to the same regression task.
+
+    This helper automatically selects a floating-point data type with
+    `_find_matching_floating_dtype` when computing with inputs that
+    follow the array API specification and ensure that all non-None inputs
+    are converted to that common dtype if needed before returning them.
+
+    Parameters
+    ----------
+    y_true : array-like
+
+    y_pred : array-like
+
+    sample_weight : array-like of shape (n_samples,)
```

Comment:
For consistency sake should `y_true` and `y_pred` also have the "of shape .." annotation?

Reply:
I realise I copied the docstring from `_check_reg_targets` without reviewing it. I'll correct both. Thanks for catching that!